lib/sinon/util/fake_server.js

```diff
@@ -21,7 +21,29 @@ function responseArray(handler) {
     return response;
 }
 
-var wloc = typeof window !== "undefined" ? window.location : { "host": "localhost", "protocol": "http"};
+function getDefaultWindowLocation() {
+    return { "host": "localhost", "protocol": "http" };
+}
+
+function getWindowLocation() {
+    if ( typeof window !== "undefined") {
+        if (typeof window.location !== "undefined") {
+            // Browsers place location on window
+            return window.location;
+        } else if ((typeof window.window !== "undefined") && (typeof window.window.location !== "undefined")) {
```

Comment:
Yes, for reasons unknown to me, this does indeed seem to be where `location` is stored, in the versions of React Native that I tested against.